pkg/asset/cluster/tfvars.go

```diff
@@ -36,24 +40,76 @@ func (t *TerraformVariables) Dependencies() []asset.Asset {
 		&installconfig.InstallConfig{},
 		&bootstrap.Bootstrap{},
 		&machine.Master{},
+		&machines.Master{},
 	}
 }
 
 // Generate generates the terraform.tfvars file.
 func (t *TerraformVariables) Generate(parents asset.Parents) error {
 	installConfig := &installconfig.InstallConfig{}
-	bootstrap := &bootstrap.Bootstrap{}
-	master := &machine.Master{}
-	parents.Get(installConfig, bootstrap, master)
+	bootstrapIgnAsset := &bootstrap.Bootstrap{}
+	masterIgnAsset := &machine.Master{}
+	mastersAsset := &machines.Master{}
+	parents.Get(installConfig, bootstrapIgnAsset, masterIgnAsset, mastersAsset)
 
-	bootstrapIgn := string(bootstrap.Files()[0].Data)
+	bootstrapIgn := string(bootstrapIgnAsset.Files()[0].Data)
 
-	masterIgn := string(master.Files()[0].Data)
+	masterIgn := string(masterIgnAsset.Files()[0].Data)
 
-	data, err := tfvars.TFVars(installConfig.Config, bootstrapIgn, masterIgn)
+	var mastersList metav1.List
+	err := yaml.Unmarshal(mastersAsset.MachinesRaw, &mastersList)
+	if err != nil {
+		return errors.Wrap(err, "unmarshal master machine list")
+	}
+
+	masters := make([]*clusterapi.Machine, 0, len(mastersList.Items))
+	for _, item := range mastersList.Items {
+		var master clusterapi.Machine
+		err = yaml.Unmarshal(item.Raw, &master)
+		if err != nil {
+			return errors.Wrap(err, "unmarshal master machine item")
+		}
+		masters = append(masters, &master)
+	}
+
+	var vpcCIDR string
+	if installConfig.Config.Platform.AWS != nil {
+		vpcCIDR = installConfig.Config.Platform.AWS.VPCCIDRBlock
+	}
+
+	var bridge string
+	var image string
+	if installConfig.Config.Platform.Libvirt != nil {
+		vpcCIDR = installConfig.Config.Platform.Libvirt.Network.IPRange
+		bridge = installConfig.Config.Platform.Libvirt.Network.IfName
+		if installConfig.Config.Platform.Libvirt.DefaultMachinePlatform == nil {
+			return errors.New("libvirt image support currently requires defaultMachinePlatform")
+		}
+		image = installConfig.Config.Platform.Libvirt.DefaultMachinePlatform.Image
+	}
+
+	var cloud string
+	if installConfig.Config.Platform.OpenStack != nil {
+		vpcCIDR = installConfig.Config.Platform.OpenStack.NetworkCIDRBlock
+		cloud = installConfig.Config.Platform.OpenStack.Cloud
+	}
+
+	data, err := tfvars.TFVars(
```

Comment:
This function is combining options from multiple platforms.. can we just switch to platform based ones?

Reply:
> ... can we just switch to platform based ones?

Sure.  In #556 I have separate assets for the per-platform Terraform variables (like [this][1]).  But #556 requires one asset per file and allows me to dynamically declare per-platform dependencies (like [this][2]).  How did you want me to fit that into the current asset structure?  Should I add another file to `TerraformVariables` or create a new asset?  Should I use a platform-specific filename like I do in #556 (e.g. see [the generated variables][3]) or a generic name like `platform.auto.tfvars`?

[1]: https://github.com/openshift/installer/blob/71bd435b2c2e9ae8885c7dc21797bf2aef634c66/pkg/installerassets/aws/terraform.go
[2]: https://github.com/openshift/installer/blob/71bd435b2c2e9ae8885c7dc21797bf2aef634c66/pkg/installerassets/cluster.go#L30
[3]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/556/pull-ci-openshift-installer-master-e2e-aws/1843/artifacts/e2e-aws/installer/terraform/aws-terraform.auto.tfvars

Reply:
we already do that for Machines, switch on platform and call platform specific TFVars. we don't need another asset.